specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,264 @@
+}
+
+/// Event args for `LaunchingRegisteredProtocol` event.
+[uuid(fc43b557-9713-4a67-af8d-a76ef3a206e8), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingRegisteredProtocolEventArgs: IUnknown {
+  /// The uri of the requested protocol.
+
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The host may set this flag to cancel the protocol launch.  If set to
+  /// `TRUE`, the protocol will not be launched. If cancelled, the
+  /// dialog is not displayed regardless of the `Handled` property.
+
+  [propget] HRESULT Cancel([out, retval] BOOL* cancel);
+
+  /// Sets the `Cancel` property.
+
+  [propput] HRESULT Cancel([in] BOOL cancel);
+
+  /// The host may set this flag to `TRUE` to hide the dialog.
+  /// The protocol request will continue as normal if it is not
+  /// cancelled, although there will be no default UI shown. By default the
+  /// value is `FALSE` and the default registered protocol dialog is shown.
+
+  [propget] HRESULT Handled([out, retval] BOOL* handled);
+
+  /// Sets the `Handled` property.
+
+  [propput] HRESULT Handled([in] BOOL handled);
+
+  /// The host may set this flag to `TRUE` to revoke previous permissions given
+  /// to this protocol/origin. In the case in which the registered protocol
+  /// launch is being requested from a trustworthy origin, the dialog
+  /// prompt (if not handled) will display a checkbox that gives the user the option to
+  /// always allow this origin to open links of this type in the associated app.
+  /// Once checked by the user, this protocol for this origin will be launched automatically
+  /// without displaying the dialog prompt.
+  /// If the `Cancel` property is set to `TRUE` while this property is set to
+  /// `TRUE` the registered protocol will not be launched in this instance. If `Handled` is set  
+  /// to `TRUE` the dialog prompt will still be disabled regardless of this property, but the 
+  /// permissions will be revoked in the future if this property is set to `TRUE`. 
+
+  [propget] HRESULT RevokeProtocolPermissionsPerOrigin(
+      [out, retval] BOOL* revokeProtocolPermissionsPerOrigin);
+
+  /// Sets the `RevokeProtocolPermissionsPerOrigin` property.
+  [propput] HRESULT RevokeProtocolPermissionsPerOrigin([in] BOOL revokeProtocolPermissionsPerOrigin);
+}
+
+``` 
+## .NET and WinRT
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2LaunchingRegisteredProtocolEventArgs;
+    
+    runtimeclass CoreWebView2LaunchingRegisteredProtocolEventArgs
+    {
+        // CoreWebView2LaunchingRegisteredProtocolEventArgs members
+        String Uri { get; };
+        Boolean Cancel { get; set; };
+        Boolean Handled {get; set; };
+        Boolean RevokeProtocolPermissionsPerOrigin {get; set; };
+    }
+
+    runtimeclass CoreWebView2
+    {
+        // CoreWebView2 
+        event Windows.Foundation.TypedEventHandler<CoreWebView2, CoreWebView2LaunchingRegisteredProtocolEventArgs> LaunchingRegisteredProtocolEventArgs;
```

Comment:
For changes to runtimeclasses we have interface_name blocks to ensure we don't break compat.  If you run the build-apiwriter.ps1 tool you can copy paste out of the generated microsoft.web.core.webview2.idl file. (also applies to the CoreWebView2Frame changes)